### ISSUE TITLE
New Bridge.$jsonGuard issue with Chrome 74.0.3729.131 update(?)

### ISSUE THREAD

**Issue by yctri** [bridgedotnet/Bridge.Newtonsoft.Json#153]
We've noticed the following issue since Chrome's 74.0.3729.131 update. 

TL;DR: I'm not sure if it's Chrome, Bridge.Newtonsoft.Json, or our code, but it appears that Bridge.$jsonGuard isn't being properly deleted. 

I added console.logs starting at line 362 of JsonConvert.js:

```
console.log("obj === object");
var removeGuard = Newtonsoft.Json.JsonConvert.defaultGuard;
if (!Bridge.$jsonGuard) {
	console.log("new jsonGuard.");
	Bridge.$jsonGuard = [];
	removeGuard = function () {
		console.log("remove jsonGuard.");
		let res = delete Bridge.$jsonGuard;
		console.log("res: " + res);
		console.log("guard (after deleting): " + Bridge.$jsonGuard);
	};

	console.log("new len: " + Bridge.$jsonGuard.length);
	console.log("new guard: " + Bridge.$jsonGuard);
}

let indexof = Bridge.$jsonGuard.indexOf(obj);

if (indexof > -1) {
	console.log("guard: " + Bridge.$jsonGuard);
	console.log("guard len: " + Bridge.$jsonGuard.length);
	console.log("indexof: " + indexof);
	console.log("indexof2: " + Bridge.$jsonGuard.indexOf(obj));
	console.log("type: " + type.toString());
	console.log("obj: " + obj.toString());
        console.dir(obj);
	return;
}
```

Console Output:

```
SerializeObject()...
...
remove jsonGuard.
res: true
guard (after deleting): undefined
(object successfully serialized)
...
SerializeObject()...
obj === object
new jsonGuard.
new len: 1024 // this should be 0
new guard: [object Object],[object Object], (etc.) // this should be empty
guard: [object Object],[object Object], (etc.) // this should be empty
guard len: 0 // correct
indexof: 0 // this should be -1
indexof2: -1 // correct
type: function () {
                this.$initialize();
                MyClass.ctor.call(this, null);
            }
obj: [object Object]
obj: (a valid object) 
SerializeObject() returns undefined.
```

The result of this error is that calling SerializeObject() will sometimes fail. Calling it 3 times in a row on the same source object succeeded the first time and failed the next 2 times.

Changing the sequence of events will offset the problem to a different object. Eventually, the problem goes away, usually after waiting a few seconds or minutes (not sure how often garbage collection runs in chrome).

The problem does not occur in firefox and only seems to occur on larger (~100k) objects.

The issue goes away as soon as I step into bridge code, hence my old-school console.log attack.

I'm not sure how our code could create this problem, given the single-threaded environment. I tried renaming $jsonGuard to make sure there was no variable conflict.

Setting $jsonGuard to null instead of deleting it makes the problem go away.

**Comment by geoffreymcgill** [comment]
@yctri – Thanks for the report. We will investigate.
